sdk/src/main/java/io/opentelemetry/sdk/trace/SpanInterceptor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import io.opentelemetry.trace.Span;
+
+/**
+ * SpanInterceptor is the interface {@code TracerSdk} uses to allow synchronous hooks for when a
+ * {@code Span} is started or when a {@code Span} is ended.
+ */
+public interface SpanInterceptor {
+  /**
+   * Called when a {@link io.opentelemetry.trace.Span} is started, if the {@link
+   * Span#isRecordingEvents()} returns true.
+   *
+   * @param span the {@code SpanSdk} that just started.
+   */
+  void onStartSync(SpanSdk span);
+
+  /**
+   * Called when a {@link io.opentelemetry.trace.Span} is ended, if the {@link
+   * Span#isRecordingEvents()} returns true.
```

Comment:
Would it be possible to create zPages as an interceptor? Do we need non-recorded spans for zPages?

Reply:
Yes.

Reply:
so zPages will not need to show non-recorded spans?

Reply:
We don't need to show non-recorded, because for these spans we don't have any information to display.